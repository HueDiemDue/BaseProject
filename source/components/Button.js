import React from 'react'
import {
  Text,
  TouchableOpacity,
  StyleSheet,
} from 'react-native'
import _ from 'lodash'
import Colors from '../styles/colors'

const Button = props => {
  const { text, style, onPress } = props

  return (
    <TouchableOpacity
      style={[styles.container, style]}
      onPress={onPress}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  )
}

Button.defaultProps = {
  value: '',
  style: null,
  text: 'Press me',
  onPress: null
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.green,
    padding: 24,
    alignItems: 'center'
  },
  text: {
    fontSize: 12,
    color: Colors.white,
    textAlign: 'center'
  }
})

export default Button