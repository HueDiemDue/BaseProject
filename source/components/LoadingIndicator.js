import React from 'react'
import {
  ActivityIndicator,
  Platform
} from 'react-native'
import _ from 'lodash'
import Colors from '../styles/colors'
import CommonStyles from '../styles/styles'

export default LoadingIndicator = props => {
  const { size, animating, center } = props

  const color = _.get(props, 'color') || (Platform.OS === 'ios' ? null : Colors.green)
  const style = center ? { ...CommonStyles.position_absolute_full, alignSelf: 'center' } : { margin: 20 }

  return (
    <ActivityIndicator
      style={style}
      color={color}
      size={size}
      animating={animating}

    />
  )
}
