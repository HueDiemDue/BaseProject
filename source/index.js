
import { Navigation } from 'react-native-navigation'
import { AppState } from 'react-native'
import { registerScreens } from './configs/screens-register'
import { defaultOptions } from './navigator/options'
import { setRootSplashScreen, setRootHomeScreen } from './navigator'

registerScreens()

Navigation.events().registerAppLaunchedListener(async () => {
  await Navigation.setDefaultOptions(defaultOptions)

  setRootSplashScreen()
})

Navigation.events().registerComponentDidAppearListener(({ componentId, componentName }) => {
  console.log('registerComponentDidAppearListener: ' + componentId + " " + componentName)
});

Navigation.events().registerComponentDidDisappearListener(({ componentId, componentName }) => {
  console.log('registerComponentDidAppearListener: ' + componentId + " " + componentName)
});