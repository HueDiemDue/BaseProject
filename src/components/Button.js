import React from 'react'
import {
  Text,
  TouchableOpacity,
  StyleSheet,
} from 'react-native'
import _ from 'lodash'
import Colors from '../styles/colors'

const Button = props => {
  const { title, style, onPress } = props

  const onPressButton = () => {
    onPress && onPress()
  }

  return (
    <TouchableOpacity
      style={[styles.container, style]}
      onPress={onPressButton}>
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  )
}

Button.defaultProps = {
  value: '',
  style: null,
  title: 'Press me',
  onPress: null
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.green,
    padding: 24,
    alignItems: 'center'
  },
  // text: {
  //   fontSize: 12,
  //   color: Colors.white,
  //   textAlign: 'center'
  // }
})

export default Button