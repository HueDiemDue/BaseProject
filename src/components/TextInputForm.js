import React from 'react'
import {
  View,
  TextInput,
  StyleSheet,
} from 'react-native'
import _ from 'lodash'
import Colors from '../styles/colors'

const TextInputForm = props => {
  const { value, style, onChangeText } = props

  return (
    <View style={[styles.container, style]}>
      <TextInput
        value={value}
        onChangeText={onChangeText}
        underlineColorAndroid={false}
        
      />
    </View>
  )
}

TextInputForm.defaultProps = {
  value: '',
  style: null,
  onChangeText: null,
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 24,
    borderColor: Colors.green,
    borderWidth: 1,
  },
  text: {
    fontSize: 12,
    color: Colors.white
  }
})

export default TextInputForm