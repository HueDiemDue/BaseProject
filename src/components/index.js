export { default as LoadingIndicator } from './LoadingIndicator'
export { default as Button } from './Button'
export { default as TextInputForm } from './TextInputForm'
