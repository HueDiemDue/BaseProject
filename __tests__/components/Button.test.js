import renderer from 'react-test-renderer'
import React from 'react'
import { Button } from '../../source/components'

describe('Components > Button', () => {
  it('Render a snapshot for Button use renderer', () => {
    const tree = renderer.create(<Button />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})