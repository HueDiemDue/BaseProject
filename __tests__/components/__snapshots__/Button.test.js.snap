// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Button Render a snapshot for Button use renderer 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "green",
      "opacity": 1,
      "padding": 24,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 12,
        "textAlign": "center",
      }
    }
  >
    Press me
  </Text>
</View>
`;
